allennlp/predictors/predictor.py

```diff
@@ -96,6 +185,15 @@ def predict_instance(self, instance: Instance) -> JsonDict:
         outputs = self._model.forward_on_instance(instance)
         return sanitize(outputs)
 
+    def predictions_to_labeled_instances(self,
+                                         instance: Instance,
+                                         outputs: Dict[str, np.ndarray]) -> List[Instance]:
```

Comment:
The idea behind this function: take the output (or predictions) of a model on an `Instance`, and make those predictions into labels or targets on the `Instance`.  This is so you can compute gradients of _what the model predicted_.  Sometimes the model predicts multiple things (like for NER, it predicts multiple spans), so we could split those predictions into separate labels so that we can compute gradients for each span separately.

So what the name should ideally capture is "turn the predictions that I got into labels on the instance".  Just `add_labels` seems a bit misleading / not descriptive enough to me.  Any other ideas?

(And Eric, feel free to take the first paragraph here and use it to improve the docstring.)

Reply:
what about like, `label_instances_using_predictions` or something like this?